content/_code-samples/normalize-currency-codes/js/normalize-currency-code.js

```diff
@@ -0,0 +1,65 @@
+////////////////////////////////////////////////////////////////////////////////
+// Normalize Currency Codes
+// Original authors: Ali (XUMM), nixer89
+// Upstream version: https://github.com/nixer89/xrpl-services-frontend/blob/master/src/app/utils/normalizers.ts
+////////////////////////////////////////////////////////////////////////////////
+// Convert from a string from either the XRP Ledger's "standard" 3-character
+// format, or several possible decodings of the "non-standard" 40-character
+// hexadecimal format into a string for humans to read.
+//
+// This code is intended for use with Node.js.
+////////////////////////////////////////////////////////////////////////////////
+
+function convertDemurrageToUTF8(demurrageCode) {
+
+    let bytes = Buffer.from(demurrageCode, "hex")
+    let code = String.fromCharCode(bytes[1]) + String.fromCharCode(bytes[2]) + String.fromCharCode(bytes[3]);
+    let interest_start = (bytes[4] << 24) + (bytes[5] << 16) + (bytes[6] <<  8) + (bytes[7]);
+    let interest_period = bytes.readDoubleBE(8);
+    const year_seconds = 31536000; // By convention, the XRP Ledger's interest/demurrage rules use a fixed number of seconds per year (31536000), which is not adjusted for leap days or leap seconds
+    let interest_after_year = Math.pow(Math.E, (interest_start+year_seconds - interest_start) / interest_period)
+    let interest = (interest_after_year * 100) - 100;
+
+    return(`${code} (${interest}% pa)`)
+}
+
+function normalizeCurrencyCode(currencyCode, maxLength = 20) {
```

Comment:
Since this sample is about `normalizeCurrencyCode` should we put this at the top? I was surprised when reading to see another function which has a different intent than the filename first.

Reply:
Demurrage-charged currencies have the interest rate encoded in their currency code, so parsing that rate and including it in the human-readable string is part of normalization.

That said, I think it would be fine to reorder the functions anyway